### DISCUSSION TITLE
errors in Vitest

### DISCUSSION THREAD

**Discussion by wangjunxian** [element-plus/element-plus#12963] [Q&A]
when I use Vitest to do test. 
it said formEl.validate is not a function.
it's ok to use it in my vue.
I don't know how to config it

**Comment by ryuhangyeong** [comment]
Maybe the link below will help you?

https://github.com/element-plus/element-plus/blob/dev/packages/components/form/__tests__/form.test.tsx#L473
https://github.com/element-plus/element-plus/blob/dev/packages/components/form/__tests__/form.test.tsx#L535

**Reply by wangjunxian** [reply]
appreciated for your help. I have been searching this problem for three days, and still can't find anything helpfull. and I saw other people have the some problems too. 
it's really little information about how to use elementPlus in Vitest.
I saw the code in the link .  but it still have the error of 'wrapper.vm.$refs.validationDataRef.validate is not a function'

I'm very sorry for bothering you again. But I do really need help. Please

this is part of my vue code:
----------------------------------------------------------
const validationDataRef = ref<FormInstance>()
const submitForm = (formEl: FormInstance | undefined) => {
  if (!formEl) return
  formEl.validate((valid, fields) => {
    if (valid) {
      login()
    } else {
    }
  })
}
----------------------------------------------------------
    <body style="background-color: #f2f2f2">
      <el-form
        ref="validationDataRef"
        :model="user"
        :rules="rules"
        status-icon
       >
----------------------------------------------------------
and this is part of my test code:
----------------------------------------------------------
  const wrapper=mount(LoginPage，{
  props :{
  user:{
   userid: "12345678",
   password: "12345678",
  }
  }
  })
function validate(){
  return(wrapper.vm.$refs.validationDataRef as FormInstance).validate() 
  .then(() => ({ valid: true, fields: undefined}))
  .catch((fields) => ({ valid: false, fields}))
}
await wrapper.vm.$nextTick()
 const res=await validate()
----------------------------------------------------------
thank you sooooo much

**Reply by ryuhangyeong** [reply]
When you print the wrapper to the console, do $refs exist normally?

**Reply by wangjunxian** [reply]
when I print wrapper.vm. it print out nothing but {}.
when I print wrapper.vm.$refs. it print out {  validationDataRef: HTMLUnknownElement { _listeners: {}, ..........
it seems like that the vue file and the test file are in the same file in which you show me.
but my test file and vue file are in different files . so I can't get the instance of vue code in my test file.  I guess.

it also have some warnings about [Failed to resolve component: el-input],[Failed to resolve component: el-form-item]  etc.

**Reply by wangjunxian** [reply]
I have solved this problem by adding a plugin.
thank you very much.

**Reply by ryuhangyeong** [reply]
I'm glad your problem is resolved.

thank you!

**Reply by wangjunxian** [reply]
sorry for that I can't paste the code  in detail. 
but this is the part of code how I solved it.
    const wrapper = mount(login, {
       global: {
        plugins: [ElementPlus],
      },
    })
hope it can help others who have the same problem with me.
